akka-http-tests/src/test/scala/akka/http/scaladsl/marshallers/sprayjson/JsonStreamingSpec.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.scaladsl.marshallers.sprayjson
+
+import akka.http.scaladsl.marshallers.Employee
+import akka.http.scaladsl.marshalling._
+import akka.http.scaladsl.model.{ ContentTypes, StatusCodes }
+import akka.http.scaladsl.server._
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import akka.http.scaladsl.unmarshalling.FromEntityUnmarshaller
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import org.scalatest.{ Matchers, WordSpec }
+import spray.json.{ DefaultJsonProtocol, JsObject }
+
+class JsonStreamingSpec extends WordSpec with Matchers with ScalatestRouteTest
+  with Directives with SprayJsonSupport {
+
+  implicit val jsonRenderingMode = JsonSourceRenderingMode.LineByLine
+
+  object EmployeeJsonProtocol extends DefaultJsonProtocol {
+    implicit val employeeFormat = jsonFormat5(Employee.apply)
+  }
+  import EmployeeJsonProtocol._
+
+  implicit def marshaller: Marshaller[Employee, ByteString] = SprayJsonSupport.sprayByteStringMarshaller[Employee]
+  implicit def unmarshaller: FromEntityUnmarshaller[Employee] = SprayJsonSupport.sprayJsonUnmarshaller[Employee]
+
+  val PostJson = (path: String, data: String) ⇒
+    Post(path).withEntity(ContentTypes.`application/json`, data)
+
+  val lineByLineCommaSeparated =
+    """|{"fname":"Frank","name":"Smith","age":42,"id":1337,"boardMember":false},
+       |{"fname":"Bob","name":"Smith","age":42,"id":1337,"boardMember":false},
+       |{"fname":"Hank","name":"Smith","age":42,"id":1337,"boardMember":false}
+    """.stripMargin
+
+  val lineByLine =
+    """|{"fname":"Frank","name":"Smith","age":42,"id":1337,"boardMember":false}
+       |{"fname":"Bob","name":"Smith","age":42,"id":1337,"boardMember":false}
+       |{"fname":"Hank","name":"Smith","age":42,"id":1337,"boardMember":false}
+    """.stripMargin
+
+  val arrayPretty =
+    """|[
```

Comment:
This isn't a valid array in JSON, missing commas, but maybe there is something special about streaming arrays?